### PR TITLE
Modernize repo

### LINKED ISSUE #1: Add SSR Support
### Discussed in https://github.com/AlaskaAirlines/WC-Generator/discussions/457

<div type='discussions-op-text'>

<sup>Originally posted by **blackfalcon** January  9, 2024</sup>
As of Lit 2.0 it's been understood that LitElement web components will be supported inside common SSR environments. This discussion is for supporting the various updates needed to deliver this capability. </div>

## Scope

The scope of this issue is to update the repo with the following criteria 

1. install the auro-library as root dependency 
1. add `type="module"` to the `./package.json`
1. `commitlint.config.js → commitlint.config.cjs`
1. delete the following local scripts 
    * . /packageScripts/postinstall.(m)js
    * ./scripts/generateDocs.(m)js
    *  ./scripts/postCss.(m)js
    *  ./scripts/pre-commit.(m)js
1.  Update the `./package.json` `"scripts"` references 

```json
"build:docs": "node ./node_modules/@aurodesignsystem/auro-library/scripts/build/generateDocs.mjs",
"preCommit": "node ./node_modules/@aurodesignsystem/auro-library/scripts/build/pre-commit.mjs",
"postCss:component": "node ./node_modules/@aurodesignsystem/auro-library/scripts/build/postCss.mjs",
"postinstall": "node ./node_modules/@aurodesignsystem/auro-library/scripts/build/postinstall.mjs",
```
### Exit criteria 

This issue can be closed once the local files have been deleted, the dependencies and scripts have been updated and the rebuilds successfully off of the auro-library resources. 

All local dependencies should be up to date as well.

### LINKED ISSUE #2: Implement theming support
### Is your feature request related to a problem? Please describe.

This is regarding theming updates that need to happen to nearly all `auro` components

### Describe the solution you'd like

1. All tokens with the prefix `--auro` need to be renamed to `--ds` so they work with the new WCSS v5 update.
2. The WCSS package needs to be updated to ^5.0.0 and the import paths in the component *.scss files need to be updated as well

### Describe alternatives you've considered

None

### Additional context

Reference this `auro-button` pr https://github.com/AlaskaAirlines/auro-button/pull/231
And this `auro-hyperlink` pr https://github.com/AlaskaAirlines/auro-hyperlink/pull/185

### Exit criteria

All tokens referencing `--auro` have been updated.

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #1, #2

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Rebuild component using generator
* Add theming support
* Add SSR support
* Refactor documentation, tests and style src code

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by jason-capsule42 — APPROVED** [review]

File: docs/partials/index.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  <!-- AURO-GENERATED-CONTENT:END -->
+</div>
+
+<auro-accordion alignRight>
+  <span slot="trigger">See code</span>
+
+<!-- AURO-GENERATED-CONTENT:START (CODE:src=./../../apiExamples/custom.html) -->
+<!-- AURO-GENERATED-CONTENT:END -->
+
+</auro-accordion>
```
Comment (blackfalcon) [inline docs/partials/index.md:104]: Delete this content

File: docs/partials/index.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+<auro-accordion alignRight>
+  <span slot="trigger">See code</span>
+
+<!-- AURO-GENERATED-CONTENT:START (CODE:src=./../../apiExamples/basic.html) -->
+<!-- AURO-GENERATED-CONTENT:END -->
+
+</auro-accordion>
+
+Having a closing statement about your example helps to really complete the thought with your reader.
```
Comment (blackfalcon) [inline docs/partials/index.md:78]: None of this makes sense? Where are all the examples currently shown here? https://auro.alaskaair.com/components/auro/interruption/drawer

Comment (blackfalcon) [reply docs/partials/index.md:78]: I see that the examples that were on the demo page are now in the API page, but none of those examples line up with the API references? All the API attributes have links that should be linked to content in the page. 

I would expect something like recently done with auro-lockup.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Review by leeejune — COMMENTED** [review]
It seems like there is no padding/margin on the sides.

# Original drawer
Screenshot 2024-01-25 at 12 39 27 AM

# PR drawer
Screenshot 2024-01-25 at 12 39 37 AM

Notice the discrepancy on the sides in the two screenshots.

File: docs/partials/index.md (hunk 1 above)
Comment (jordanjones243) [reply docs/partials/index.md:104]: Per our discussion, we will leave this example in.

File: docs/partials/index.md (hunk 2 above)
Comment (jordanjones243) [reply docs/partials/index.md:78]: There have been links for the attribute examples implemented.

**Review by leeejune — APPROVED** [review]

**Pull request merged**